src/lib.rs

```diff
@@ -264,7 +264,7 @@ pub enum CargoOpt {
 /// # Parameters
 ///
 /// - `manifest_path`: Path to the manifest.
-pub fn metadata(manifest_path: Option<&Path>) -> Result<Metadata> {
+pub fn metadata<P: AsRef<Path>>(manifest_path: Option<P>) -> Result<Metadata> {
```

Comment:
can you use `Option<impl AsRef<Path>>` instead of explicit generic args?

Reply:
That would set the minimum rust version to 1.26 ...is that ok?

Reply:
Since this is only happening with the next breaking change, ppl will need to manually bump their version anyway, so seems fine.